Docs/developer_guide/mrml_overview.md

```diff
@@ -100,14 +100,346 @@ The following methods on the MRML scene are used to manage Undo/Redo stacks:
+
+#### The widget representation
+
+The widget representation is the visualization half of displaying a node on screen. This is where any data structures describing your type are turned into [`vtkActor`](https://vtk.org/doc/nightly/html/classvtkActor.html)s that can be displayed in a VTK render window.
+
+Files:
+
+* \<Extension>/\<Module>/VTKWidgets/vtkSlicer\<MyCustomType>WidgetRepresentation.h
+* \<Extension>/\<Module>/VTKWidgets/vtkSlicer\<MyCustomType>WidgetRepresentation.cxx
+
+Key Points:
+
+* Naming convention for class: `vtkSlicer<MyCustomType>WidgetRepresentation`
+* Inherits from [`vtkMRMLAbstractWidgetRepresentation`](http://apidocs.slicer.org/master/classvtkMRMLAbstractWidgetRepresentation.html).
+* Constructing a new node is same as the [data node](#the-data-node).
+* Override [`void UpdateFromMRML(vtkMRMLNode* caller, unsigned long event, void *callData = nullptr)`](http://apidocs.slicer.org/master/classvtkMRMLAbstractWidgetRepresentation.html#a404ce09ab17e92d203c82fee2a71b0ee) to update the widget representation when the underlying data or display nodes change.
+* To make the class behave like a vtkProp override:
+  * [`void GetActors(vtkPropCollection*)`](http://apidocs.slicer.org/master/classvtkMRMLAbstractWidgetRepresentation.html#ab0124c85fa2d5032a37c690d58c1ef2f)
+  * [`void ReleaseGraphicsResources(vtkWindow*)`](http://apidocs.slicer.org/master/classvtkMRMLAbstractWidgetRepresentation.html#aa1949d0b26d58d83df1c0764db6d19bf)
+  * [`int RenderOverlay(vtkViewport* viewport)`](http://apidocs.slicer.org/master/classvtkMRMLAbstractWidgetRepresentation.html#a51314976a613b4c7324a571ee7243cbc)
+  * [`int RenderOpaqueGeometry(vtkViewport* viewport)`](http://apidocs.slicer.org/master/classvtkMRMLAbstractWidgetRepresentation.html#a14e33bf29e9272dd9fe29c7baadec730)
+  * [`int RenderTranslucentPolygonalGeometry(vtkViewport* viewport)`](http://apidocs.slicer.org/master/classvtkMRMLAbstractWidgetRepresentation.html#a0d00da2fcb99df807c6884ac757e1735)
+  * [`vtkTypeBool HasTranslucentPolygonalGeometry()`](http://apidocs.slicer.org/master/classvtkMRMLAbstractWidgetRepresentation.html#a16347752784cec5542751b9e6d81cf2b)
+
+:::{important}
+
+The points/lines/etc pulled from the data node should be post-transform, if there are any transforms applied.
+
+:::
+
+:::{tip}
+
+Minimize the number of actors used for better rendering performance.
+
+:::
+
+#### The displayable manager
+
+The data node, display node, widget, and widget representation are all needed pieces for data actually showing up on the screen. The displayable manager is the glue that brings all the pieces together. It monitors the MRML scene, and when data and display nodes are added or removed, it creates or destroys the corresponding widgets and widget representations.
+
+Files:
+
+* \<Extension>/\<Module>/MRMLDM/vtkMRML\<MyCustomType>DisplayableManager.h
+* \<Extension>/\<Module>/MRMLDM/vtkMRML\<MyCustomType>DisplayableManager.cxx
```

Comment:
```
|-- <Extension>
       |-- <Module>
              |-- MRMLDM
                    |-- vtkMRML<MyCustomType>DisplayableManager.h
                    |-- vtkMRML<MyCustomType>DisplayableManager.cxx
```

Reply:
Updated to suggested file structure formatting.